pkgs/bananaPiR4/mt7988-persistent-mac-from-cpu-uid.patch

```diff
@@ -0,0 +1,76 @@
+Generate Persistent MAC Addr from CPU UID
+
+BananaPi R3 devices lack an assigned UAA MAC, and therefor on each boot
+Linux generates a new MAC address at random.  This is not good for the user,
+resulting in IPv4 and IPv6 addressess changing on every boot.
+
+Use the system CPU ID to give us a stable but unique set of bytes that we then
+hash to get more bytes.  We hash the hash when we need more bytes.
+
+diff --git a/board/mediatek/mt7988/mt7988_rfb.c b/board/mediatek/mt7988/mt7988_rfb.c
+index 846c715ca05..b1a6d11f77d 100644
+--- a/board/mediatek/mt7988/mt7988_rfb.c
++++ b/board/mediatek/mt7988/mt7988_rfb.c
+@@ -4,7 +4,62 @@
+  * Author: Sam Shih <sam.shih@mediatek.com>
+  */
+ 
++#include <common.h>
++#include <mapmem.h>
++#include <hash.h>
++#include <u-boot/sha1.h>
++#include <net.h>
++
+ int board_init(void)
+ {
+ 	return 0;
+ }
++
++int board_late_init(void)
++{
++	void *buf;
++	uint8_t cpuid_digest[20];
++	u8 cpu_uid[8];
```

Comment:
As the size of this data has grown in comparison to the MT7986 which only had 8 bytes, should we use the whole 16?

```suggestion
+	u8 cpu_uid[16];
```

And then 
```
+	buf = map_sysmem(0x11F50140, 16);
+	memcpy(cpu_uid, buf, 16);
+	unmap_sysmem(buf);
+
+	sha1_csum(cpu_uid, 16, cpuid_digest);
```